graph/cytoscape/cytoscape.go

```diff
@@ -261,41 +254,85 @@ func buildConfig(trafficMap graph.TrafficMap, nodes *[]*NodeWrapper, edges *[]*E
-	httpIn := getRate(s.Metadata, "httpIn")
-
-	if httpIn > 0.0 {
-		nd.HttpIn = fmt.Sprintf("%.2f", httpIn)
-
-		httpIn3xx := getRate(s.Metadata, "httpIn3xx")
-		httpIn4xx := getRate(s.Metadata, "httpIn4xx")
-		httpIn5xx := getRate(s.Metadata, "httpIn5xx")
-
-		if httpIn3xx > 0.0 {
-			nd.HttpIn3xx = fmt.Sprintf("%.2f", httpIn3xx)
-		}
-		if httpIn4xx > 0.0 {
-			nd.HttpIn4xx = fmt.Sprintf("%.2f", httpIn4xx)
+func addNodeTelemetry(n *graph.Node, nd *NodeData) {
+	nd.Traffic = []ProtocolTraffic{}
+	for _, p := range graph.Protocols {
+		protocolTraffic := ProtocolTraffic{Protocol: p.Name}
+		for _, r := range p.NodeRates {
+			if rateVal := getRate(n.Metadata, r.Name); rateVal > 0.0 {
+				if protocolTraffic.Rates == nil {
+					protocolTraffic.Rates = make(map[string]string)
+				}
+				protocolTraffic.Rates[r.Name] = fmt.Sprintf("%.*f", r.Precision, rateVal)
+			}
 		}
-		if httpIn5xx > 0.0 {
-			nd.HttpIn5xx = fmt.Sprintf("%.2f", httpIn5xx)
+		if protocolTraffic.Rates != nil {
+			nd.Traffic = append(nd.Traffic, protocolTraffic)
 		}
 	}
+}
 
-	httpOut := getRate(s.Metadata, "httpOut")
-
-	if httpOut > 0.0 {
-		nd.HttpOut = fmt.Sprintf("%.2f", httpOut)
+func addEdgeTelemetry(e *graph.Edge, ed *EdgeData) {
+	if val, ok := e.Metadata["isMTLS"]; ok {
+		ed.IsMTLS = val.(bool)
 	}
-
-	tcpIn := getRate(s.Metadata, "tcpIn")
-	tcpOut := getRate(s.Metadata, "tcpOut")
-
-	if tcpIn > 0.0 {
-		nd.TcpIn = fmt.Sprintf("%.2f", tcpIn)
+	if val, ok := e.Metadata["responseTime"]; ok {
+		responseTime := val.(float64)
+		ed.ResponseTime = fmt.Sprintf("%.0f", responseTime)
 	}
-	if tcpOut > 0.0 {
-		nd.TcpOut = fmt.Sprintf("%.2f", tcpOut)
+	if val, ok := e.Source.Metadata["isUnused"]; ok {
+		ed.IsUnused = val.(bool)
+	}
+
+	ed.Traffic = []ProtocolTraffic{}
+	for _, p := range graph.Protocols {
+		protocolTraffic := ProtocolTraffic{Protocol: p.Name}
+		total := 0.0
+		err := 0.0
+		var percentErr, percentReq graph.Rate
+		for _, r := range p.EdgeRates {
+			rateVal := getRate(e.Metadata, r.Name)
+			switch {
+			case r.IsTotal:
+				total = rateVal
+			case r.IsErr:
+				err += rateVal
+			case r.IsPercentErr:
+				percentErr = r
+			case r.IsPercentReq:
+				percentReq = r
+			}
+			if rateVal := getRate(e.Metadata, r.Name); rateVal > 0.0 {
+				if protocolTraffic.Rates == nil {
+					protocolTraffic.Rates = make(map[string]string)
+				}
+				protocolTraffic.Rates[r.Name] = fmt.Sprintf("%.*f", r.Precision, rateVal)
+			}
+		}
+		if protocolTraffic.Rates != nil {
+			if total > 0 {
+				if percentErr.Name != "" {
+					rateVal := err / total * 100
+					if rateVal > 0.0 {
+						protocolTraffic.Rates[percentErr.Name] = fmt.Sprintf("%.*f", percentErr.Precision, rateVal)
+					}
+				}
+				if percentReq.Name != "" {
+					rateVal := 0.0
+					for _, r := range p.NodeRates {
+						if !r.IsOut {
+							continue
+						}
+						rateVal = total / getRate(e.Source.Metadata, r.Name) * 100.0
+						break
+					}
+					if rateVal > 0.0 && rateVal < 100.0 {
```

Comment:
You don't want this if rateVal == 100? (though using equality with floating point isn't usually valid, but, seems odd you don't include this if its 100.

Reply:
We don't currently include 100% in the JSON because that value is assumed if not provided.  It's the same as not including a 0 rate.

Reply:
I'm pretty sure UXD told us not to do that - they want to see 100%

Reply:
You will see 100%, but it's done on the UI side.